test/expr.jl

```diff
@@ -0,0 +1,538 @@
+macro audit(expr::Expr)
+    result = quote
+        tuple(
+            try
+                @eval let
+                    $expr
+                end
+            catch e
+                e
+            end,
+            $(QuoteNode(expr)),
+        )
+    end
+    return esc(result)
+end
+
+macro expr(expr::Expr)
+    esc(QuoteNode(expr))
+end
+
+function strip_lineno!(expr::Expr)
+    filter!(expr.args) do ex
+        isa(ex, LineNumberNode) && return false
+        if isa(ex, Expr)
+            ex.head === :line && return false
+            strip_lineno!(ex::Expr)
+        end
+        return true
+    end
+    return expr
+end
+
+macro test_splitdef_invalid(expr)
+    result = quote
+        @test_throws ArgumentError splitdef($expr)
+        @test splitdef($expr, throw=false) === nothing
+    end
+    return esc(result)
+end
+
+@testset "splitdef / combinedef" begin
```

Comment:
If the objective is to keep the same API as MacroTools then it might be good to make MacroTools a test dependency and also compare all the cases that should be the same and explicitly show which cases MacroTools won't handle.

Reply:
I'm not sure that would be worth the effort. The functions here definitely don't have the same API (note `:type` is never returned by MacroTools.splitdef). I'll update the description to mention what isn't handled by MacroTools